### PR TITLE
Fix 11476. Watch node logs instead of grepping.

### PR DESCRIPTION
**Pull request by ptnapoleon**
Solves problem where query is aborted but error hasn't been flushed
to log yet

### CODE REVIEW

File: pushed_notifications_test.py [hunk 1]
```diff
@@ -313,9 +313,9 @@ def read_failure_query():
 
         read_failure_query()
 
-        failure = (node1.grep_log(failure_msg) or
-                   node2.grep_log(failure_msg) or
-                   node3.grep_log(failure_msg))
+        failure = (node1.watch_log_for(failure_msg, timeout=5) or
```
Comment (mambocab) [inline pushed_notifications_test.py]: Does this work when we raise an exception? https://github.com/pcmanus/ccm/blob/master/ccmlib/node.py#L425

Comment (ptnapoleon) [inline pushed_notifications_test.py]: Nope. But, if you look down ten lines, it wouldn't work there either. I'll grep after watching

Comment (mambocab) [inline pushed_notifications_test.py]: I don't follow.

**Comment by mambocab** [comment]
Nice. +1 based on visual inspection; want me to run it?

**Comment by ptnapoleon** [comment]
If you'd like. It passes locally.

**Comment by mambocab** [comment]
Naw. +1.

**Pull request merged**